src/librustc_trans/type_of.rs

```diff
@@ -178,7 +178,13 @@ pub fn in_memory_type_of<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>, t: Ty<'tcx>) ->
           }
           let llet = in_memory_type_of(cx, e);
           let n = t.simd_size(cx.tcx()) as u64;
-          Type::vector(&llet, n)
+          // see comment in abi.rs where we set `x86_mmx` to true for why we
+          // compare to 1 here and return a different type.
```

Comment:
Oh, I'd expect this to have the "main copy" of the comment and `abi.rs` to point to it. Not a big deal though. The same thing about checking the size applies here too I guess.